lib/commons/blocs/preferences_bloc.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+// ----- EVENTS -----
+
+class PreferencesEvent extends Equatable {
+  @override
+  List<Object> get props => [];
+}
+
+class PreferencesLoadEvent extends PreferencesEvent {}
+
+class PreferencesSaveEvent extends PreferencesEvent {
+  final bool useAutomaticTheme;
+  final bool useDarkTheme;
+
+  PreferencesSaveEvent({this.useAutomaticTheme, this.useDarkTheme});
+}
+
+// ----- STATES -----
+
+class PreferencesState extends Equatable {
+  @override
+  List<Object> get props => [];
+}
+
+class PreferencesUninitializedState extends PreferencesState {}
+
+class PreferencesLoadingState extends PreferencesState {}
+
+class PreferencesSavingState extends PreferencesState {}
+
+class PreferencesUpdatedState extends PreferencesState {
+  final bool useAutomaticTheme;
+  final bool useDarkTheme;
+
+  PreferencesUpdatedState({@required this.useAutomaticTheme, @required this.useDarkTheme});
+}
+
+// ----- BLOC -----
+
+class PreferencesBloc extends Bloc<PreferencesEvent, PreferencesState> {
+  SharedPreferences _preferences;
```

Comment:
Should be remove ! By using a attributes, you're caching the preferences: if any part of the code modify this `SharedPreferences` this bloc won't be aware of that change... => Never store a state in the bloc.

In the future, we **must** create a data provider for the preferences (not in this PR but the next time we want to store a preference)